deepmd/dpmodel/model/make_model.py

```diff
@@ -377,4 +385,94 @@ def _format_nlist(
             assert ret.shape[-1] == nnei
             return ret
 
+        def do_grad_r(
+            self,
+            var_name: Optional[str] = None,
+        ) -> bool:
+            """Tell if the output variable `var_name` is r_differentiable.
+            if var_name is None, returns if any of the variable is r_differentiable.
+            """
+            return self.atomic_model.do_grad_r(var_name)
+
+        def do_grad_c(
+            self,
+            var_name: Optional[str] = None,
+        ) -> bool:
+            """Tell if the output variable `var_name` is c_differentiable.
+            if var_name is None, returns if any of the variable is c_differentiable.
+            """
+            return self.atomic_model.do_grad_c(var_name)
+
+        def serialize(self) -> dict:
+            return self.atomic_model.serialize()
+
+        @classmethod
+        @abstractmethod
+        def deserialize(cls, data) -> "CM":
+            pass
```

Comment:
Any idea of implementing the class method.  in principle, we only need to deserialize the `self.atomic_model`

Reply:
self.atomic_model.deserialize returns a BaseAtomicModel, but we want a BaseModel.

Reply:
Oh, I come up with a trick way to do so.